src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/CallGCSuppressFinalizeCorrectlyTests.cs

```diff
@@ -135,6 +135,43 @@ End Sub
             await VerifyVB.VerifyAnalyzerAsync(code);
         }
 
+        [Fact]
+        public async Task AsyncDisposableWithFinalizer_CSharp_NoDiagnostic()
+        {
+            var code = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+class MyAsyncDisposable : IAsyncDisposable
+{
+    [DllImport(""example.dll"")]
+    private static extern int GetHandle();
+
+    [DllImport(""example.dll"")]
+    private static extern void FreeHandle(int handle);
+
+    private readonly int handle;
+
+    public MyAsyncDisposable()
+    {
+        this.handle = GetHandle();
+    }
+
+    ~MyAsyncDisposable()
+    {
+        FreeHandle(this.handle);
+    }
+
+    public async ValueTask DisposeAsync()
```

Comment:
@bartonjs Sorry to bother you but is there any guideline as the one for `Dispose` with the `virtual void Dispose(bool disposing)` pattern for the async version?

Reply:
```C#
public async ValueTask DisposeAsync()
{
    await DisposeAsyncCore().ConfigureAwait(false);
    Dispose(false);
    GC.SuppressFinalize(this);
}

protected virtual ValueTask DisposeAsyncCore()
{
    // Async version of code inside `if (disposing)` blocks from Dispose(bool)
}
```

Reply:
@mavasani It might be worth having a separate issue and update the comply to dispose pattern to support this new pattern. WDYT?

Reply:
See #3909